File: Source/Triggers/SampleTrigger.vb
Imports Celeste.[Mod].Entities
Imports Microsoft.Xna.Framework

<CustomEntity("VeryBasicMod/SampleTrigger")>
Public Class SampleTrigger
    Inherits Trigger
    Public Sub New(data As EntityData, offset As Vector2)
        MyBase.New(data, offset)
        ' TODO: read properties from data
    End Sub

End Class

File: Source/Triggers/SampleTrigger.fs
namespace Celeste.Mod.VeryFunnyMod.Triggers

open Celeste.Mod
open Celeste
open System
open Microsoft.Xna.Framework
open Celeste.Mod.Entities

[<CustomEntity("VeryFunnyMod/SampleTrigger")>]
type SampleTrigger(data: EntityData, offset: Vector2) as this =
    inherit Trigger(data, offset)
